CONTRIBUTING.adoc

```diff
@@ -373,6 +402,42 @@ The job will:
 3. Compare the two runs.
 4. Post the comparison as a comment to the PR.
 
+=== Merging PRs
+
+==== Merge method and commit history
+
+All 3 Github merge methods (merge commit, squash and merge, and rebase and merge) are allowed.
+Use whichever you deem appropriate.
+As said before, sometimes people use a single PR with multiple commits for their work; other times they create multiple small PRs.
+The best merge method is different for different cases.
+
+That being said, there are not many cases where "rebase and merge" is appropriate.
+And if you use this method, your PR must have a clean commit history: every commit should have a meaningful message, and should be buildable.
+You don't want to have commits like "fix a typo", "this may work" or "wip, done for the day" in master with a linear history.
+And if some of these commits are non-buildable, it can create problems for "git bisect".
+
+This is slightly less of a problem when you use the "merge commit" method.
+While these interim commits would still be unpleasant, at least the merge commits and the non-linear history clearly indicate where they come from.
+
+The best thing to do, of course, is to not have those interim commits.
+If you think merging multiple commits makes more sense, clean up the history.
+If you don't, squash. The option chosen can vary from PR to PR.
+
+==== Beware divergence of master and PR branch
+
+Merging a PR can break master, if the PR branch has diverged from master, even if CI on the PR is green.
```

Comment:
This is discussed a bit in the "CI" section as "semantic" merge conflicts. Maybe delete that since this covers it?